R/homoSapiens/humanAtlas.r

```diff
@@ -58,7 +58,33 @@ appendGenes <- function(humanGenesVector, GSEMatrix)
   return(fullMatrix) #Return fullMatrix
 }
 
+checkRawCounts <- function(GSEMatrix)
+{
+  GSEBigMatrix <- as.big.matrix(GSEMatrix)
+  maxValue <- max(as.numeric(unlist(GSEMatrix)))
+  if (maxValue > 50)
+  {
+    return("log normalized")
+  }
+  else if (is.float(GSEMatrix))
```

Comment:
using `is.float` is fine but I think you can do this easier with built in functions in base R. Here are some ideas

``` r

# integer matrix
x_int = matrix(0L, 5, 5)
typeof(x_int)
#> [1] "integer"
is.integer(x_int)
#> [1] TRUE
is.numeric(x_int)
#> [1] TRUE
is.double(x_int)
#> [1] FALSE

# numeric double precision matrix
x_dbl = matrix(1.0, 5, 5)
typeof(x_dbl)
#> [1] "double"
is.integer(x_dbl)
#> [1] FALSE
is.numeric(x_dbl)
#> [1] TRUE
is.double(x_dbl)
#> [1] TRUE

check_vals <- function(x, max_log_val = 50) {
  if(is.integer(x)){
    return("raw counts")
  } else if (is.double(x)) {
    
    if(max(x) > max_log_val){
      return("normalized")
    } else if (min(x) < 0) {
      stop("negative values detected, likely scaled data")
    } else {
      return("log-normalized")
    }
  } else {
    stop("unknown matrix format: ", typeof(x))
  }
}


check_vals(x_dbl)
#> [1] "log-normalized"
check_vals(x_int)
#> [1] "raw counts"
check_vals(matrix(51.010910, 5, 5))
#> [1] "normalized"

check_vals(matrix(-1.010910, 5, 5))
#> Error in check_vals(matrix(-1.01091, 5, 5)): negative values detected, likely scaled data
check_vals(matrix(TRUE, 5, 5))
#> Error in check_vals(matrix(TRUE, 5, 5)): unknown matrix format: logical
check_vals(matrix(1i, 5, 5))
#> Error in check_vals(matrix(0+1i, 5, 5)): unknown matrix format: complex
check_vals(matrix("Hello", 5, 5))
#> Error in check_vals(matrix("Hello", 5, 5)): unknown matrix format: character
check_vals(matrix(factor("Hello"), 5, 5))
#> Error in check_vals(matrix(factor("Hello"), 5, 5)): unknown matrix format: character
```

<sup>Created on 2020-07-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Yeah I could try an approach like this since it will just use base R functions and I won't need to use other R libraries. Will get on this right away @kriemo.

Reply:
See commit https://github.com/rnabioco/scRNA-seq-Cell-Reference-Matrix/pull/19/commits/020060bd3937512cf1d9dffa4697bf06cc15bb2e